### DISCUSSION TITLE
merge_messages is not public?

### DISCUSSION THREAD

**Discussion by TheVova** [protobuf-c/protobuf-c#602] [Q&A] [answered]
Hi, i was looking through the source code, when i noticed this function, merge_messages in protobuf-c.c, which is marked static and isnt used anywhere in the code i could find. This seems to do the same as MergeFrom in the cpp API, and is then a very useful function... what is the reason it is static and unused?

**Comment by lipnitsk** [comment, marked as answer]
`merge_messages` is used internally by protobuf-c and is not designed to be used from outside (for example, it corrupts the earlier message, which may not be what the caller wants).

**Comment by TheVova** [comment]
I see, but surely as long as that behaviour is documented (and it is very well stated in the comment above it) it is up to the caller to decide what they want? 
again, the stated behaviour of this function matches MergeFrom, which is a very useful function to have. Would we be interested in opening this up as an available api, and if so what would be required to in order to do so? more tests? docs? i can help :)
